src/test/kotlin/org/rust/lang/core/pattern/RustPatternTestBase.kt

```diff
@@ -0,0 +1,13 @@
+package org.rust.lang.core.pattern
+
+import com.intellij.patterns.ElementPattern
+import org.rust.lang.RustTestCaseBase
+
+abstract class RustPatternTestBase : RustTestCaseBase() {
+    override val dataPath: String get() = ""
+
+    protected fun <T>testPattern(code: String, pattern: ElementPattern<T>) {
```

Comment:
With the latest Kotlin plugin (update via `Tools > Kotlin > Configure Kotlin plugin updates`) you should be able to get rid of `//language=Rust` comments and instead use an annotation:

```
protected fun <T>testPattern(@Language("Rust") code: String, pattern: ElementPattern<T>) {
```

See https://github.com/intellij-rust/intellij-rust/blob/069cd1d05c3f567d32e40010c80a4a2dee7ecc83/src/test/kotlin/org/rust/ide/search/RustFindUsagesTest.kt for an example.